resources/snakefiles/Snakefile.smk

```diff
@@ -0,0 +1,242 @@
+
+samples=sample_table.index
+reads=['R1', 'R2']
+
+def get_read(sample, read):
+    return(sample_table.loc[sample, read])
+
+rule all:
+    input:
+        "output/qc/multiqc/multiqc.html",
+        "output/metaspades/S22207_S103.contigs.fasta",
+        "output/metaspades/S22205_S104.contigs.fasta",
+
+rule fastqc_pre_trim:
+    input:
+        lambda wildcards: get_read(wildcards.sample,
+                                   wildcards.read)
+    output:
+        html="output/qc/fastqc/pre_trim/{sample}.{read}.html",
+        zip="output/qc/fastqc/pre_trim/{sample}.{read}_fastqc.zip" # the suffix _fastqc.zip is necessary for multiqc to find the file. If not using multiqc, you are free to choose an arbitrary filename
+    params: ""
+    threads:
+        config['threads']['fastqc']
+    wrapper:
+        "0.72.0/bio/fastqc"
+
+rule cutadapt_pe:
+    input:
+        lambda wildcards: sample_table.loc[wildcards.sample,
+                                      'R1'],
+        lambda wildcards: sample_table.loc[wildcards.sample,
+                                      'R2']
+    output:
+        fastq1="output/trimmed/{sample}.R1.fastq.gz",
+        fastq2="output/trimmed/{sample}.R2.fastq.gz",
+        qc="output/logs/cutadapt/{sample}.qc.txt"
+    params:
+        "-a {} {}".format(config["params"]["cutadapt"]['adapter'],
+                          config["params"]["cutadapt"]['other'])
+    threads:
+        config['threads']['cutadapt_pe']
+    wrapper:
+        "0.17.4/bio/cutadapt/pe"
+
+rule fastqc_post_trim:
+    input:
+        "output/trimmed/{sample}.{read}.fastq.gz"
+    output:
+        html="output/qc/fastqc/post_trim/{sample}.{read}.trimmed.html",
+        zip="output/qc/fastqc/post_trim/{sample}.{read}.trimmed_fastqc.zip" # the suffix _fastqc.zip is necessary for multiqc to find the file. If not using multiqc, you are free to choose an arbitrary filename
+    params: ""
+    threads:
+        config['threads']['fastqc_post_trim']
+    wrapper:
+        "0.72.0/bio/fastqc"
+
+
+rule host_filter:
+    """
+    Performs host read filtering on paired end data using Bowtie and Samtools/
+    BEDtools.
+
+    Also requires an indexed reference (path specified in config).
+
+    First, uses Bowtie output piped through Samtools to only retain read pairs
+    that are never mapped (either concordantly or just singly) to the indexed
+    reference genome. Fastqs from this are gzipped into matched forward and
+    reverse pairs.
+
+    Unpaired forward and reverse reads are simply run through Bowtie and
+    non-mapping gzipped reads output.
+
+    All piped output first written to localscratch to avoid tying up filesystem.
+    """
+    input:
+        fastq1=rules.cutadapt_pe.output.fastq1,
+        fastq2=rules.cutadapt_pe.output.fastq2
+    output:
+        nonhost_R1="output/filtered/nonhost/{sample}.1.fastq.gz",
+        nonhost_R2="output/filtered/nonhost/{sample}.2.fastq.gz",
+        host_R1="output/filtered/host/{sample}.1.fastq.gz",
```

Comment:
Do we want to change this to a single BAM for the output of this rule?

Reply:
This is done now.